### DISCUSSION TITLE
Automatisches Karte aktualisieren nach Geometrieverortung/-änderung

### DISCUSSION THREAD

**Discussion by wwvlbg** [e-netze/webgis-community#173] [General]
Wir verwenden das Editieren von Geometrien in einer Oracle-DB.
Das funktioniert auch wunderbar.

In WebGIS 4 sah man die neu verortete oder geänderte Geometrie unmittelbar nach dem speichern.
In WebGIS 5 sind die Geometrien erst dann sichtbar, wenn man die Karte nach dem speichern manuell aktualisiert (Button "Karte aktualisieren", Zoom in/out, Ausschnitt verschieben etc).

Kann man das irgendwo konfigurieren, dass neu verortete oder geänderte Geometrien unmittelbar nach dem speichern als solches auch sichtbar sind? Das wäre für die User wirklich hilfreich.

**Comment by wwvlbg** [comment]
Zusammen mit NÖ konnten wir das "Aktualisierungsproblem" nach dem Editieren von Geometrien eingrenzen.
Das Problem tritt offenbar dann auf, wenn in einen Geoserver-WFS-Dienst editiert wird (dieser Dienst schreibt dann bei uns in die Oracle-DB). Diverse Einstellungsversuche am Geoserver (zB Deaktivierung vom Tile Caching) haben am Verhalten leider nichts geändert.

Bestünde die Möglichkeit, eine automatische (oder konfigurierbare) Refresh-Funktion einzubauen?
Der Anwendungsfall ist der, dass der Karten-Refresh dann ausgeführt werden soll, nachdem man beim Editieren auf Speichern klickt.

**Comment by gis-eni** [comment]
WebGIS kann eigentlich keine WFS Dienste editieren?! Aber ich denke einmal die Konstellation ist folgende:

* Es gibt einen WMS Dienst, der die Geo Objekte anzeigt (Darstellungsdienst)
* Es gibt einen WFS Dienst, über den man die Geo Objekte abfragen kann (Abfragedienst)
* WebGIS schreibt direkt über Editing Parametrierung in eine Oracle Datenbank

Die Editing Parametrierung (wie immer sie auch erfolgt ist) ist dem den Abfragedienst verknüpft. Damit kann man ein Objekt auswählen und dann bearbeiten, oder? Weil grundsätzlich muss eine Geo Objekte FeatureClass auch abfragbar sein, damit man sie editieren kann.

Wenn man ein Geo Objekt dann verändert/erstellt und speichert wird der entsprechende Dienst (hier der Abfragedienst) neu gezeichnet, damit das veränderte/neue Geo Objekt in der Karte sichtbar ist.
Das Problem ist hier, dass der Dienst für die Darstellung jetzt ein anderer ist, nämlich der WMS Darstellungsdienst.

Ist das etwa sie Ausganslage?

Wenn ja, ist bei der Abfrage hier etwas eingestellt:

![image](https://github.com/gis-eni/webgis-forum/assets/62870568/c210a381-2182-44c6-81a3-ef011c3b64bc)

Unter diesem Punkt kann man Themen angeben, die mit der Abfrage verknüpft sein sollten. Also unter diesem Dienst muss man das entsprechende Darstellungsthema aus dem WMS Dienst einstellen. Damit könnte WebGIS dann erkennen, welche Dienste zusätzlich nach dem Speichern neu gezeichnet werden sollten.

Und Frage: Funktioniert das Editing in Oracle in NÖ? Und wenn ja, was machen die anders?

**Comment by wwvlbg** [comment]
Danke für die Rückmeldung!

Die skizzierte Ausgangslage ist soweit richtig!
Die Darstellungsdienste sind WMS- und die Abfragedienste sind WFS-Dienste.
Was bei uns noch dazukommt ist, dass der WFS-Dienst über den in die Oracle-DB geschrieben wird, keine Abfragethemen hat.
Das mussten wir trennen (hat diverse Gründe).
D.h. ich ich kann das mit dem WMS-Dienst als Toc Thema so leider nicht machen.

Ev. hat NÖ dieselben Dienste für Darstellung und Abfrage?

**Comment by gis-eni** [comment]
Ok, wenn das Thema, dass editiert wird nicht einmal eine Abfrage hat, versteh ich gar nicht, wie es dann editiert werden kann :/
Da empfehle ich, den Weg einzuschlagen, denn NÖ umgesetzt hat. Die haben das direkte Editieren in Oracle damals implementieren lassen (ohne Wartung). 
WebGIS unterstützt das eigentlich nur inoffiziell. Offiziell sollte editieren von Geo Objekte immer über ein Feature Service erfolgen und NIE direkt in die Datenbank.

**Comment by wwvlbg** [comment]
Wahrscheinlich habe ich mich falsch ausgedrückt... 

Wir verwenden natürlich auch die Variante von NÖ, d.h. wir editieren direkt in die Oracle DB (Konfig in der workspaces.xml bzw. Ordner themes).
Im WFS ist ja eigentlich nur konfiguriert in welches edittheme editiert wird.

**Comment by gis-eni** [comment]
Um noch einmal zurück zu kommen: In NÖ funktioniert das? Und was machen die anders?

Die Möglichkeit der automatischen Aktualisierung der Karte ist natürlich implementiert. Wenn der Dienst allerdings ein Dienst ohne Darstellung ist (WFS) muss man das dem WebGIS irgendwie sagen (siehe oben).

**Comment by wwvlbg** [comment]
Ich habe seitens NÖ folgende Infos bekommen (ev. hilft das weiter)....

<
Für das Editieren in ORACLE verwenden wir die gleiche Software wie ihr. Nur die Darstellung / Abfrage setzen wir eine andere Kartenserver Software ein.

Für die Darstellung und Abfrage ist ESRI ArcGIS Server im Einsatz. Der ArcGIS Server gliedert ein oder mehrere Kartenschichten zu einem eigenen Service. Das Prinzip ist daher grundsätzlich unterschiedlich zu GeoServer. Aber das Ergebnis sollte das gleiche sein.

Wir verwenden für die Darstellung und Abfrage 2 ArcGIS Server Dienste
•	wis_edit für die Kartenschichten, die editiert werden  wird ausschließlich in der webgis-Karte „WIS“ verwendet. Nur in dieser Karte „WIS“ werden Geometrien erstellt und verändert.
•	wis_edit_info für die Abfragen und Darstellung  wir haben pro WIS Modul eine eigene Karte zum Ansehen und Abfragen, wo Kartenschichten angezeigt werden. Jede Kartenschicht ist eine Datenbank Query. In ArcGIS heißen diese Layer „Query Layer“.

Für deinen Fall ist daher nur unsere Karte „WIS“ relevant, weil das ein vergleichbarer Anwendungsfall ist.
1.	Person steigt in WIS-Anwendung (von Jürgen Schaffer) ein
2.	In der WIS-Anwendung werden alle Attribute verwaltet, in webgis5-Karte ausschließlich die Geometrie. 
In der WIS-Anwendung wird ein neues Objekt samt Attribute angelegt
3.	Es fehlt noch die Geometrie 
  daher wird aus der WIS-Anwendung per Link die webgis5-Karte „WIS“ aufgerufen. Die WIS-ID des neuen WIS-Objekts wird mit der URL mitübergeben.
4.	Die webgis5-Karte öffnet sich:
In der Karte wird nur diese WIS-ID dargestellt. Wir verwenden hier im webgis5 Darstellungsfilter, welche meines Wissens nach nur mit ArcGIS Server in webgis5 konfiguriert werden können. Alle anderen WIS-Objekte sind ausblendet.
5.	In der webgis5-Karte wird die Geometrie zur übergebenen WIS-Id eingezeichnet und gespeichert.

Unterschiede können folglich nur durch den Karten Server (ArcGIS Server) oder dem Darstellungsfilter verursacht sein.
>

**Reply by gis-eni** [reply]
Aber warum parametriert man für das Edit Thema nicht nicht einfach eine Abfrage und verknüpft diese wie oben gezeigt, mit dem Thema der Darstellungsdienstes. In WebGIS 4 war dieser Mechanismus eigentlich der gleiche.

**Reply by wwvlbg** [reply]
Ich habe das jetzt so versucht.
Der gewünschte Effekt bleibt leider aus, d.h. keine sofortige Anzeige der Geometrieänderungen.

Wenn das mit dem automatischen Refresh nicht funktioniert/machbar ist, dann lassen wir es gut sein...

**Comment by gis-eni** [comment]
Nachdem es auch mit der empfohlen Methode auch nicht funktioniert hat, habe ich den Thread einmal als "Bug" markiert. Sollte dann mit dem nächsten Release behoben sein.
